examples/fabric/image_classifier/train_fabric.py

```diff
@@ -0,0 +1,198 @@
+            # NOTE: no need to call `.to(device)` on the data, target
+            optimizer.zero_grad()
+            output = model(data)
+            loss = F.nll_loss(output, target)
+            fabric.backward(loss)  # instead of loss.backward()
+
+            optimizer.step()
+            if (batch_idx == 0) or ((batch_idx + 1) % hparams.log_interval == 0):
+                print(
+                    "Train Epoch: {} [{}/{} ({:.0f}%)]\tLoss: {:.6f}".format(
+                        epoch,
+                        batch_idx * len(data),
+                        len(train_loader.dataset),
+                        100.0 * batch_idx / len(train_loader),
+                        loss.item(),
+                    )
+                )
+                if hparams.dry_run:
+                    break
+
+        scheduler.step()
+
+        # TESTING LOOP
+        model.eval()
+        test_loss = 0
+        with torch.no_grad():
+            for data, target in test_loader:
+                # NOTE: no need to call `.to(device)` on the data, target
+                output = model(data)
+                test_loss += F.nll_loss(output, target, reduction="sum").item()
+
+                # WITHOUT TorchMetrics
+                # pred = output.argmax(dim=1, keepdim=True)  # get the index of the max log-probability
+                # correct += pred.eq(target.view_as(pred)).sum().item()
+
+                # WITH TorchMetrics
+                test_acc(output, target)
+
+                if hparams.dry_run:
+                    break
+
+        # all_gather is used to aggregated the value across processes
+        test_loss = fabric.all_gather(test_loss).sum() / len(test_loader.dataset)
+
+        print(f"\nTest set: Average loss: {test_loss:.4f}, Accuracy: ({100 * test_acc.compute():.0f}%)\n")
+        test_acc.reset()
+
+        if hparams.dry_run:
+            break
+
+    # When using distributed training, use `fabric.save`
+    # to ensure the current process is allowed to save a checkpoint
+    if hparams.save_model:
+        fabric.save(model.state_dict(), "mnist_cnn.pt")
+
+
+if __name__ == "__main__":
+    # Arguments can be passed in through the CLI as normal and will be parsed here
+    # Example:
+    # lightning run model image_classifier.py accelerator=cuda --epochs=3
+    parser = argparse.ArgumentParser(description="Fabric MNIST Example")
+    parser.add_argument(
+        "--batch-size", type=int, default=64, metavar="N", help="input batch size for training (default: 64)"
+    )
+    parser.add_argument("--epochs", type=int, default=14, metavar="N", help="number of epochs to train (default: 14)")
+    parser.add_argument("--lr", type=float, default=1.0, metavar="LR", help="learning rate (default: 1.0)")
+    parser.add_argument("--gamma", type=float, default=0.7, metavar="M", help="Learning rate step gamma (default: 0.7)")
+    parser.add_argument("--dry-run", action="store_true", default=False, help="quickly check a single pass")
```

Comment:
why dry-run in example ?

Reply:
dry-run is used to quickly check single pass of training loop. It is used in the lightning example as well.